### PR TITLE
Corrected physics query max result checking

### LINKED ISSUE #17464: Collide_shape max results buffer error
**Godot version:**
Godot 3.0.2 stable

**OS/device including version:**
Windows 10, not really applicable

**Issue description:**
If there are more collisions then max_results specified in the function collide_shape:
http://docs.godotengine.org/en/3.0/classes/class_physicsdirectspacestate.html#class-physicsdirectspacestate-collide-shape

Godot crashes. Not sure if its ignoring the max but the buffer isn't large enough or if it just tried to add one to many. 

@AndreaCatania you might want to have a look at this

**Steps to reproduce:**
This is reproducible in my teleporter test by editing the teleport function script and setting the max collisions to 1, it will crash when it finds more then 1 collider to collide with.

**Minimal reproduction project:**
https://www.dropbox.com/s/2wvmgddglkkg3oc/Test_teleport.zip?dl=0

### PR DESCRIPTION
**Pull request by AndreaCatania**
Fixes #17464 
can be cherry picked

### CODE REVIEW

**Comment by LikeLakers2** [comment]
Without having too much experience with what you're doing, I'm wondering why this is keeping a return value at all, since the return value isn't ever used in Bullet. Though I also wonder why it's being changed to always return 1 if you're not going to outright remove it, since people with their own forks might be using that return value?

I know forks aren't really a concern, I'm just wondering why you're changing it to always return 1 instead of either leaving it be or removing the return.

**Comment by AndreaCatania** [comment]
The function is virtual and its definition returns a float number, so I need to return something.
So why am I returning 1 instead of 0? Because this return value is meant to be used internally to test if the next element should be added as result or not depending of its penetration.
So here I'm returning 1 so if in a future it will be used the behaviour doesn't change since all objects will be returned no matter the penetration.

Initially I thought that this value was used since the docs say so, but now by checking the code and the bullet forum I understood that it's not yet used so I changed how I get results.

File: modules/bullet/godot_result_callbacks.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/*************************************************************************/
+﻿/*************************************************************************/
```
Comment (akien-mga) [inline modules/bullet/godot_result_callbacks.cpp]: You also added a BOM here.

**Comment by AndreaCatania** [comment]
@akien-mga I've removed BOM

**Comment by hpvb** [comment]
Cherry picked into 3.0.3

**Pull request merged**